folium/plugins/marker.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+"""
+Marker plugin
+--------------
+
+"""
+from jinja2 import Template
+import json
+
+from .plugin import Plugin
+from .links import JavascriptLink
+
+class Popup(Plugin):
+    def __init__(self, html, max_width=300):
+        """TODO : docstring here"""
+        super(Popup, self).__init__()
+        self.plugin_name = 'Popup'
+        self.html = json.dumps(html)
+        self.max_width = max_width
+        self.map = None
+
+    def render(self, **kwargs):
+        """TODO : docstring here"""
+        assert self.map is not None, "Cannot render a Popup that have no 'map' attribute."
+        self.map.figure.script['Popup_'+self.object_name] = Template("""
+        var Popup_{id} = L.popup({{
+            maxWidth: '{maxWidth}'
+            }}).setContent({html});""".format(
+                html=self.html,
+                maxWidth=self.max_width,
+                id=self.object_name,
+                ))
+        return 'Popup_'+self.object_name
+
+class VegaPopup(Plugin):
+    def __init__(self, data, width=300, height=300):
+        """TODO : docstring here"""
+        super(VegaPopup, self).__init__()
+        self.plugin_name = 'VegaPopup'
+        self.data = data
+        self.map = None
+        self.width = "{}px".format(width) if isinstance(width,int) or isinstance(width,float) else "{}".format(width)
+        self.height = "{}px".format(height) if isinstance(height,int) or isinstance(height,float) else "{}".format(height)
+
+    def render(self, **kwargs):
+        """TODO : docstring here"""
+        assert self.map is not None, "Cannot render a Popup that have no 'map' attribute."
+        self.map.figure.header['d3'    ] = JavascriptLink("https://cdnjs.cloudflare.com/ajax/libs/d3/3.5.5/d3.min.js")
+        self.map.figure.header['vega'  ] = JavascriptLink("https://cdnjs.cloudflare.com/ajax/libs/vega/1.4.3/vega.min.js")
+        self.map.figure.header['jquery'] = JavascriptLink("https://code.jquery.com/jquery-2.1.0.min.js")
+        self.map.figure.script['vega_parse'] = Template("""function vega_parse(spec, div) {
+            vg.parse.spec(spec, function(chart) { chart({el:div}).update(); });}""")
+
```

Comment:
With this approach it should be relatively easy to create mpld3 (pr even bokeh) popups.